swarm/pss/notify/notify.go

```diff
@@ -0,0 +1,276 @@
+package notify
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/swarm/pss"
+)
+
+const (
```

Comment:
can you explain why this whole thing is not a devp2p protocol over pss?
seems like the textbook example. instead we choose to implement the same things.
what am i missing?

Reply:
because it's not a one-to-one connection.

That is to say, the notifications aren't one-to-one.

The control messages, however, are reminiscient of protocol (start, stop, notifywithkey). B considering start and stop to be within the same conceptual "connection" is quite the stretch; you could be connected to the resource for a year before you stop. That seems more like individual RPC calls to me.

Reply:
ok right.